### DISCUSSION TITLE
How to import and use different fonts

### DISCUSSION THREAD

**Discussion by LFernandoMB** [rust-skia/rust-skia#942] [General]
Hello friends, I'm probably doing something very stupid, but I'm trying to use different fonts in my application and I'm not having positive results.

```rust
fn draw_static_text(
        &mut self,
        content: &str,
        translation: Translation,
        size: f32,
        color: skia_safe::Color4f,
    ) {
        let mut font_collection = FontCollection::new();

        let font_file_path = Path::new("src/skia/digital-7.ttf");

        let file_bytes = File::open(&font_file_path)?;

        let mut buffer = [0; 1024];
        let bytes_read = file_bytes.read(&mut buffer)?;
        let bytes_slice = &buffer[..bytes_read];

        let font_manager = FontMgr::new_from_data(bytes_slice, None, None)?;

        font_collection.set_asset_font_manager(font_manager);

        // font_collection.set_default_font_manager(FontMgr::new(), None);
        let paragraph_style = ParagraphStyle::new();
        let mut paragraph_builder = ParagraphBuilder::new(&paragraph_style, self.font_collection());
        let mut binding = TextStyle::new();
        let ts = binding.set_font_size(size);

        ts.set_foreground_paint(Paint::new(color, None).set_anti_alias(true));
        paragraph_builder.push_style(ts);
        paragraph_builder.add_text(content);

        let mut paragraph = paragraph_builder.build();
        paragraph.layout(600.0);
        paragraph.paint(
            self.get_surface().canvas(),
            Point {
                x: translation[0],
                y: translation[1],
            },
        );
    }
``` 

I just want to read the "digital-7.ttf" file and use the font to write the text. Can you help me?

Note: I didn't include all the errors that are generating because maybe there is another way to do this, but if it is "correct" and you need the errors, I can provide them.

**Comment by pragmatrix** [comment]
Since the latest changes in the font management, this got a lot more difficult and quite counter-intuitive. The problem is that only the default font FontMgr (which is the system's FontMgr created with FontMgr::new()) is able to load custom fonts, but there is no possibility to add these fonts to it, sooo you - as I can see - you have to load typefaces with the system's FontMgr and then store it in a TypefaceFontProvider which itself can be converted to a FontMgr. Here is an example I've added to the skia-org's example program in the PR #945: https://github.com/rust-skia/rust-skia/blob/2978c8f2ed389062c8c3d2a31fa8d4e9ebb6c0c7/skia-org/src/skparagraph_example.rs#L30-L57 

I hope this helps.

**Comment by LFernandoMB** [comment]
Thank you very much friend, it worked! Worked perfectly.
